src/main/scala/io/lenses/data/generator/schema/DatasetCreator.scala

```diff
@@ -0,0 +1,75 @@
+package io.lenses.data.generator.schema
+
+import cats.effect.IO
+import io.lenses.data.generator.http.LensesClient
+import org.http4s.client.blaze.BlazeClientBuilder
+import org.http4s.Uri
+import scala.concurrent.ExecutionContext
+import cats.effect.ContextShift
+import org.http4s.client.Client
+import org.http4s.Request
+import org.http4s.Method
+import org.http4s.circe._
+import io.circe.Json
+import org.http4s.Status
+import io.lenses.data.generator.cli.LensesCreds
+
+trait DatasetCreator {
+  def create(name: String, schema: Schema)(implicit
+      ec: ExecutionContext,
+      cs: ContextShift[IO]
+  ): IO[Unit]
+}
+
+object DatasetCreator {
+  def Kafka(
+      lensesClient: LensesClient,
+      lensesCreds: LensesCreds
+  ): DatasetCreator =
+    new DatasetCreator {
+      override def create(schemaName: String, schema: Schema)(implicit
```

Comment:
Why not use a T for which we can have a schema for. See Avro4s.

Reply:
Let me clarify: the generator I am working on is specifically intended to produce random schemas (not records) to facilitate testing the Explore search. The approach I have taken is the following:

- define a simple, canonical representations of a schema.
- use scalacheck to automatically generate an arbitrary number of random instances of these canonical schema
- convert the canonical schema into the datasource specific representation (e.g. AVRO, Elastic type mappings, Postgres DLLs).
- Create the schema directly in the source data system (currently, this means creating empty topics/indices/tables).

While this is the main focus of the story I am working on, we will definitely need to implement also other generators that produce random records for a given schema. As you suggest, these might be simply driven by classes (any `T`) and use avro4s.